### PR TITLE
Fix restart state loop

### PR DESCRIPTION
**Pull request by imsickofmaps**
Using the special state maker for itself which causes loops on timeout on timeout.

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@justinvdm any chance of a review on this - you're the one who knows most about the timeout state - I've deployed this in QA to check this was the root cause and it is.

FYI - @smn

File: test/personal.test.js [hunk 1]
```diff
@@ -2103,6 +2103,70 @@ describe("app", function() {
...
+                        .setup.user.addr("27001")
+                        .setup.user.lang(null)
+                        .setup.user.answers({})
+                        .setup.user.metadata({
+                            session_length_helper: {
+                                start:1415791076582
+                            }
+                        })
+                        .setup.user.state('states_language')
+                        .inputs({session_event: 'new'})
```
Comment (justinvdm) [inline test/personal.test.js:2126]: How do we figure out if a timeout happened? I'm probably forgetting something, but a single new session event doesn't seem like enough information to know that a timeout happened.

Comment (justinvdm) [inline test/personal.test.js:2126]: Ah, nvm, reading `go.utils.timeout` made me remember why this works.

**Comment by hodgestar** [comment]
:+1: from me but Justin is still reviewing.

File: test/personal.test.js [hunk 2]
```diff
@@ -2103,6 +2103,70 @@ describe("app", function() {
...
+                            state: 'states_timed_out',
+                            reply: [
+                                'Welcome back. Please select an option:',
+                                '1. Continue signing up for messages',
+                                '2. Main menu'
+                            ].join('\n')
+                        })
+                        .run();
+                });
+            });
```
Comment (justinvdm) [inline test/personal.test.js:2169]: This test looks the same as the previous test, anything I'm missing?

**Comment by justinvdm** [comment]
Minor comment, looks good otherwise.

File: test/personal.test.js (hunk 2 above)
Comment (imsickofmaps) [inline test/personal.test.js:2169]: One setup user state is a standard state, this one it's the states_timed_out state. The root cause of the loop, timing out on a timeout :cry:

Comment (justinvdm) [inline test/personal.test.js:2169]: Oh, right, sorry missed that.

**Pull request merged**